src/parse.rs

```diff
@@ -107,10 +110,46 @@ fn fields_from_ast(
     fields: &syn::punctuated::Punctuated<syn::Field, Token![,]>,
 ) -> Result<Vec<Field>> {
     // serde::internals::ast.rs:L183
+    let mut idx = 0;
     fields
         .iter()
-        .enumerate()
-        .map(|(i, field)| {
+        .map(|field| {
+            let mut skip_serializing_if = Skip::None;
+            for attr in &field.attrs {
+                if attr.path().is_ident("serde") {
+                    attr.parse_nested_meta(|meta| {
+                        if meta.path.is_ident("skip_serializing_if") {
+                            let litstr: LitStr = meta.value()?.parse()?;
+                            let tokens = syn::parse_str(&litstr.value())?;
+                            if !skip_serializing_if.is_none() {
+                                return Err(meta
+                                    .error("Multiple attributes for skip_serializing_if or skip"));
+                            }
+                            skip_serializing_if = Skip::If(syn::parse2(tokens)?);
+                            Ok(())
+                        } else if meta.path.is_ident("skip") {
+                            if meta.value().is_ok() {
+                                return Err(meta.error("`skip` does not expect any value"));
+                            }
+                            if !skip_serializing_if.is_none() {
+                                return Err(meta
+                                    .error("Multiple attributes for skip_serializing_if or skip"));
+                            }
+                            skip_serializing_if = Skip::Always;
+                            Ok(())
+                        } else {
+                            Err(meta.error("Unkown field attribute"))
+                        }
+                    })?;
+                }
+            }
+
+            let index = idx;
+
+            if !matches!(skip_serializing_if, Skip::Always) {
+                idx += 1;
+            }
```

Comment:
Per the discussion in this thread, we should increment the index even if the field is skipped.

Reply:
I'll add an option to be able to not increment. (Default would be to increment).